apps/src/templates/courseOverview/AssignCourse.js

```diff
@@ -0,0 +1,194 @@
+import React, { Component, PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+import Radium from 'radium';
+import $ from 'jquery';
+import color from "@cdo/apps/util/color";
+import i18n from '@cdo/locale';
+import ProgressButton from '@cdo/apps/templates/progress/ProgressButton';
+import BaseDialog from '@cdo/apps/templates/BaseDialog';
+import AssignCourseConfirm from './AssignCourseConfirm';
+
+const styles = {
+  button: {
+    marginBottom: 10
+  },
+  icon: {
+    fontSize: 24,
+    // we want our icon text to be a different size than our button text, which
+    // requires we manually offset to get it centered properly
+    position: 'relative',
+    top: 3
+  },
+  dropdown: {
+    border: `1px solid ${color.charcoal}`,
+    position: 'absolute',
+    marginTop: -10
+  },
+  section: {
+    padding: 10,
+    color: color.charcoal,
+    backgroundColor: color.white,
+    fontFamily: '"Gotham 5r", sans-serif',
+    display: 'block',
+    textDecoration: 'none',
+    lineHeight: '20px',
+    transition: 'background-color .2s ease-out',
+    ':hover': {
+      backgroundColor: color.lightest_gray,
+      cursor: 'pointer'
+    }
+  },
+  nonFirstSection: {
+    borderTop: `1px solid ${color.charcoal}`
+  }
+};
+
+/**
+ * A dropdown component that shows a list of sections for the current user, and
+ * lets them assign the current course to any of those sections (after accepting
+ * a confirmation dialog)
+ */
+class AssignCourse extends Component {
+  static propTypes = {
+    courseId: PropTypes.number.isRequired,
+    courseName: PropTypes.string.isRequired,
+    sectionsInfo: PropTypes.arrayOf(PropTypes.shape({
+      id: PropTypes.number.isRequired,
+      name: PropTypes.string.isRequired,
+    })).isRequired,
+  };
+
+  state = {
+    dropdownOpen: false,
+    sectionIndexToAssign: null,
+    errorString: null
+  }
+
+  expandDropdown = () => {
+    document.addEventListener('click', this.onClickDocument, false);
+    this.setState({dropdownOpen: true});
+  }
+
+  collapseDropdown = () => {
+    document.removeEventListener('click', this.onClickDocument, false);
+    this.setState({dropdownOpen: false});
```

Comment:
You might still need to handle the componentWillUnmount case here, in case we unmount while the dropdown is open.